cmd/crane/cmd/serve.go

```diff
@@ -37,11 +37,11 @@ func newCmdRegistry() *cobra.Command {
 }
 
 func newCmdServe() *cobra.Command {
-	var disk bool
+	var address, disk string
 	cmd := &cobra.Command{
 		Use:   "serve",
 		Short: "Serve an in-memory registry implementation",
-		Long: `This sub-command serves an in-memory registry implementation on an automatically chosen port (or $PORT)
+		Long: `This sub-command serves an in-memory registry implementation on an automatically chosen port (:0), $PORT or --address
```

Comment:
Does address assume it includes the `:` then? Should it always be `localhost:NNNN` ?

Reply:
oh it can't be localhost always, that was the behavior in https://github.com/google/go-containerregistry/pull/1746 but that didn't work in a container. 

address is basically can be one of these with no restrictions

`:0` (kernel chosen) 
`localhost:0` (kernel chosen port on loopback interface)
`0.0.0.0:$port` 

sky is the limit here.

Reply:
i just tried to preserve the old behavior here while allowing users to override the address string.

Reply:
`crane registry serve --disk=$(mktemp -d)` works nicely.

```
2023/11/29 12:47:34 storing blobs in /var/folders/c3/qcpmmp4s0_1cy0b3bxjwcsqh0000gn/T/tmp.B4TRj2kT
2023/11/29 12:47:34 serving on port 63643
```

Reply:
and if you do `crane registry serve --disk=$(mktemp -d) --blobs-to-disk`

```
Flag --blobs-to-disk has been deprecated, and will stop working in a future release. use --disk=$(mktemp -d) instead.
Error: --disk and --blobs-to-disk can't be used together
```